internal/pkg/agent/application/coordinator/diagnostics_test.go

```diff
@@ -0,0 +1,416 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package coordinator_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	mock "github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v2"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-client/v7/pkg/proto"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/coordinator/mocks"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/info"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/transpiler"
+	"github.com/elastic/elastic-agent/internal/pkg/config"
+	"github.com/elastic/elastic-agent/internal/pkg/diagnostics"
+	"github.com/elastic/elastic-agent/pkg/component"
+	"github.com/elastic/elastic-agent/pkg/component/runtime"
+	"github.com/elastic/elastic-agent/pkg/control/v2/cproto"
+	"github.com/elastic/elastic-agent/pkg/core/logger"
+)
+
+// Refer to https://vektra.github.io/mockery/installation/ to check how to install mockery binary
+//go:generate mockery --name ComponentsModifier
+//go:generate mockery --name ConfigChange
+//go:generate mockery --name RuntimeManager
+//go:generate mockery --name ConfigManager
+//go:generate mockery --name VarsManager
+//go:generate mockery --name UpgradeManager
+//go:generate mockery --name ReExecManager
+//go:generate mockery --name MonitorManager
+//go:generate mockery --dir ../../../capabilities/ --name Capability
+//go:generate mockery --dir ../../../core/composable/ --name FetchContextProvider
+
+var /*const*/ expectedDiagnosticHooks map[string]string = map[string]string{
+	"pre-config":      "pre-config.yaml",
+	"variables":       "variables.yaml",
+	"computed-config": "computed-config.yaml",
+	"components":      "components.yaml",
+	"state":           "state.yaml",
+}
+
+var /*const*/ linuxPlatformDetail component.PlatformDetail = component.PlatformDetail{
+	Platform: component.Platform{
+		OS:   "linux",
+		Arch: "amd64",
+		GOOS: "linux",
+	},
+	Family: "",
+	Major:  "22",
+	Minor:  "4",
+}
+
+func TestCoordinatorDiagnosticHooks(t *testing.T) {
```

Comment:
I missed reviewing this before it was merged, but it looks great. Thanks @pchila for sticking with this.